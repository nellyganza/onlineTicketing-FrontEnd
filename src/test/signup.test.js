import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount} from 'enzyme';
import store from '../store';
import App from '../app';
import Signup from '../pages/signup';
import TopNav from '../components/common/AuthTopNav';
import Dashboard from '../pages/DashBoard';

describe('Testing Signup ', () => {
  it('should render dashbaord', () => {
    const wrapper = renderer.create(<Provider store={store}><Router><Dashboard /></Router></Provider>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('top navigation  should be rendered', () => {
    const wrapper = renderer.create(<Provider store={store}><Router><TopNav /></Router></Provider>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('Should fillup and submit the signup form', () => {
    const wrapper = mount(<Provider store={store}><Router><Signup /></Router></Provider>);
    const firstName = wrapper.find('#firstName');
    firstName.simulate('change', { target: { name: 'firstName', value: 'umuhire' } });
    const lastName = wrapper.find('#lastName');
    lastName.simulate('change', { target: { name: 'lastName', value: 'umuhire' } });
    const email = wrapper.find('#email');
    email.simulate('change', { target: { name: 'email', value: 'umuhire@test.com' } });
    const password = wrapper.find('#Password');
    password.simulate('change', { target: { name: 'password', value: 'umuhire' } });
    const ConfirmPassword = wrapper.find('#confirmPassword');
    ConfirmPassword.simulate('change', { target: { name: 'confirmPassword', value: 'umuhire' } });
    const button = wrapper.find('#submitButton');
    button.simulate('click');
    const form = wrapper.find('form');
    form.simulate('submit');
  });
});
