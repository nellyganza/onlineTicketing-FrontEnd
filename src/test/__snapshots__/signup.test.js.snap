// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Signup  should render dashbaord 1`] = `
<div
  className="w-screen overflow-hidden"
>
  <div
    className="w-screen"
  >
    <div
      className="w-full float-right fixed right-0 z-10"
    >
      <div>
        <nav
          className="flex justify-between bg-formColor w-full z-10  p-3 md:py-3 md:pl-4  md:pr-6 h-full border-b border-gray-100"
        >
          <a
            className="block"
            href="/"
            onClick={[Function]}
          >
            <span
              className="text-xl text-white font-bold  font-Sansation italic"
            >
              BareFoot
            </span>
          </a>
          <div>
            <div
              className="flex space-x-6 mt-2"
            >
              <div>
                <div>
                  <svg
                    className="w-4 fill-current text-black"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="flex text-sm space-x-2 cursor-pointer"
                onClick={[Function]}
              >
                <div>
                  <svg
                    className="w-4 fill-current text-black"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
                    />
                  </svg>
                </div>
                <span
                  className="text-white"
                />
              </div>
            </div>
          </div>
        </nav>
        <div
          className="bg-formColor text-white absolute right-6 py-2 px-4 hidden"
        >
          <ul>
            <li>
              Plofile
            </li>
            <li>
              Logout
            </li>
          </ul>
        </div>
      </div>
    </div>
    <main
      className="mt-14 p-4 absolute full left-0 max-h-screen text-gray-700"
    />
  </div>
</div>
`;

exports[`Testing Signup  top navigation  should be rendered 1`] = `
<div>
  <nav
    className="flex justify-between bg-formColor w-full z-10  p-3 md:py-3 md:pl-4  md:pr-6 h-full border-b border-gray-100"
  >
    <a
      className="block"
      href="/"
      onClick={[Function]}
    >
      <span
        className="text-xl text-white font-bold  font-Sansation italic"
      >
        BareFoot
      </span>
    </a>
    <div>
      <div
        className="flex space-x-6 mt-2"
      >
        <div>
          <div>
            <svg
              className="w-4 fill-current text-black"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
              />
            </svg>
          </div>
        </div>
        <div
          className="flex text-sm space-x-2 cursor-pointer"
          onClick={[Function]}
        >
          <div>
            <svg
              className="w-4 fill-current text-black"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
              />
            </svg>
          </div>
          <span
            className="text-white"
          />
        </div>
      </div>
    </div>
  </nav>
  <div
    className="bg-formColor text-white absolute right-6 py-2 px-4 hidden"
  >
    <ul>
      <li>
        Plofile
      </li>
      <li>
        Logout
      </li>
    </ul>
  </div>
</div>
`;
